File: 77-combinations/combinations.py
class Solution:
    def helper(self, i, n, k, ans, slate):
        if k==len(slate):
            ans.append(slate[:])
            return 
        if i==n+1:
            return 
        slate.append(i)
        self.helper(i+1,n,k,ans,slate)
        slate.pop()
        self.helper(i+1,n,k,ans,slate)
    def combine(self, n: int, k: int) -> List[List[int]]:
        ans=[]
        slate=[]
        self.helper(1,n,k,ans,slate)
        return ans

File: 77-combinations/combinations.cpp
class Solution {
    void helper(int i, int n, int k, vector<int>&slate, vector<vector<int>>&ans){
        if (k==slate.size()){
            ans.push_back(slate);
            return;
        }
        if (i==n+1){
            return;
        }
        
        slate.push_back(i);
        helper(i+1,n,k,slate,ans);
        slate.pop_back();
        helper(i+1,n,k,slate,ans);

    }
public:
    vector<vector<int>> combine(int n, int k) {
        vector<vector<int>>ans;
        vector<int>slate;
        helper(1,n,k,slate,ans);
        return ans;
    }
};